### PR TITLE
Several partner-related functions

### PR DESCRIPTION
**Pull request by Brotenko**
Mainly decompiled Bombette, Bow, Kooper, Parakarry and Goombario related Functions. 
Some other Functions that might be vaguely related have also been decompiled.
Refactored the Actor struct.

Everything related to `MovePos` will break with this PR, but afaik @ethteck said he would fix that. There are also some issues with splat again (I think?), so this PR will still need to see some work until it can actually be merged - Sorry for that!

### CODE REVIEW

File: src/battle/partner/parakarry_6FFD80.c [hunk 1]
```diff
@@ -1,13 +1,74 @@
 #include "common.h"
+#include "battle/battle.h"
 
 INCLUDE_ASM(s32, "battle/partner/parakarry_6FFD80", func_80238000_6FFD80);
 
-INCLUDE_ASM(s32, "battle/partner/parakarry_6FFD80", func_80238C88_700A08);
+ApiStatus func_80238C88_700A08(ScriptInstance* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    u8 level = battleStatus->partnerActor->staticActorData->level;
```
Comment (nanaian) [inline src/battle/partner/parakarry_6FFD80.c:8]: this temp likely not required, can just go in the switch?

File: src/battle/partner/parakarry_6FFD80.c [hunk 2]
```diff
@@ -1,13 +1,74 @@
...
+            var0 = 5;
+            break;
+        case 1:
+            var0 = 6;
+            break;
+        case 2:
+            var0 = 7;
+            break;
+        default:
+            break;
```
Comment (nanaian) [inline src/battle/partner/parakarry_6FFD80.c:22]: is this case required

File: src/battle/partner/parakarry_6FFD80.c [hunk 3]
```diff
@@ -1,13 +1,74 @@
...
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_80238CE0_700A60(ScriptInstance* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    Actor* partnerActor = battleStatus->partnerActor;
+    Actor* targetActor = get_actor(partnerActor->targetActorID);
+    ActorPart* targetActorPart = get_actor_part(targetActor, partnerActor->targetPartIndex);
+    s32 airLiftChance = targetActor->staticActorData->airLiftChance;
+    s32 temp = 100 - (targetActor->currentHP * 100) / targetActor->maxHP;
```
Comment (nanaian) [inline src/battle/partner/parakarry_6FFD80.c:35]: percentHPremaining

File: src/battle/partner/parakarry_6FFD80.c [hunk 4]
```diff
@@ -1,13 +1,74 @@
...
+    }
+
+    if (airLiftChance > 0) {
+        airLiftChance = airLiftChance + ((airLiftChance * temp) / 100);
+
+        if (airLiftChance > 100) {
+            airLiftChance = 100;
+        }
+
+        if (targetActor->debuff != 0) {
```
Comment (nanaian) [inline src/battle/partner/parakarry_6FFD80.c:52]: Debuff_NONE iirc

File: src/battle/partner/watt_703AF0.c [hunk 5]
```diff
@@ -36,4 +36,45 @@ INCLUDE_ASM(s32, "battle/partner/watt_703AF0", func_80238C08_7046F8);
 
 INCLUDE_ASM(s32, "battle/partner/watt_703AF0", func_80238C84_704774);
 
-INCLUDE_ASM(s32, "battle/partner/watt_703AF0", func_80238E5C_70494C);
+// It matches, but splat split it wrong
```
Comment (nanaian) [inline src/battle/partner/watt_703AF0.c:39]: @ethteck

File: src/battle/partner/watt_703AF0.c [hunk 6]
```diff
@@ -36,4 +36,45 @@ INCLUDE_ASM(s32, "battle/partner/watt_703AF0", func_80238C08_7046F8);
...
+#ifdef NON_MATCHING
+ApiStatus func_80238E5C_70494C(ScriptInstance* script, s32 isInitialCall) {
```
Comment (nanaian) [inline src/battle/partner/watt_703AF0.c:41]: hey its that same function again. please .inc.c this!

File: src/code_18F340.c [hunk 7]
```diff
@@ -2,7 +2,20 @@
 
 INCLUDE_ASM(s32, "code_18F340", func_80260A60);
 
-INCLUDE_ASM(s32, "code_18F340", func_80260AD4);
+//INCLUDE_ASM(s32, "code_18F340", func_80260AD4);
+ApiStatus func_80260AD4(ScriptInstance* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    Actor* playerActor = battleStatus->playerActor;
+    u8 var0 = playerActor->debuff - 3;
```
Comment (nanaian) [inline src/code_18F340.c:9]: similar story here for a u8 implying lots of ors. is this the same function again?

**Review by nanaian — CHANGES_REQUESTED** [review]

File: src/battle/partner/bow_710EF0.c [hunk 8]
```diff
@@ -1,15 +1,113 @@
...
+    Actor* partnerActor = battleStatus->partnerActor;
+    Actor* targetActor;
+    ActorDesc* targetActorDesc;
+    ActorPart* targetActorPart;
+    s32 targetActorDescBaseStatusChance;
+    s32 var0 = 0;
+    s32 var1 = 0;
+    s32 var2 = 0;
+
+    for (var2; var2 < partnerActor->targetListLength; var2++) {
```
[the remark this answers is not in the document]
Comment (ethteck) [reply src/battle/partner/bow_710EF0.c:86]: also I'd call loop iterators i

File: src/battle/partner/watt_703AF0.c (hunk 5 above)
Comment (ethteck) [reply src/battle/partner/watt_703AF0.c:39]: fixed in my next PR

File: include/functions.h [hunk 9]
```diff
@@ -255,6 +255,15 @@ void show_damage_popup(f32 x, f32 y, f32 z, s32 damageAmount);
...
 void func_80070BB0(s32 arg0, f32 arg1, f32 arg2, f32 arg3, f32 arg4, s32 arg5);
+void func_80071FF0(s32 arg0, f32 arg1, f32 arg2, f32 arg3, f32 arg4, s32 arg5);
+void func_80070970(s32 arg0, f32 arg1, f32 arg2, f32 arg3, s32 arg4, s32 arg5);
+void func_800709D0(s32 arg0, f32 arg1, f32 arg2, f32 arg3, f32 arg4, s32 arg5);
+void func_80070130(s32 arg0, f32 arg1, f32 arg2, f32 arg3);
+void func_800701F0(s32 arg0, f32 arg1, f32 arg2, f32 arg3);
+void add_xz_vec3f(Vec3f* vector, f32 speed, f32 angleDeg);
+void play_movement_dust_effects(s32 var0, f32 xPos, f32 yPos, f32 zPos, f32 angleDeg);
+
+Actor func_80072110(s32, s32, s32, s32, f32, s32); // Probably wrong
```
[the remark this answers is not in the document]
Comment (Brotenko) [reply include/functions.h:266]: This was from a Function that I stopped working on (for now at least) and no matter what it returned, it didn't seem to match. it normally has to be `Actor*`, I agree with that, but it somehow broke everything.
For now this is just there so I don't forget about it, but I can also just remove it since it's wrong.

[the remark this answers is not in the document]
Comment (nanaian) [reply include/functions.h:266]: yeah remove it then

File: src/battle/partner/bombette_6FAD10.c [hunk 10]
```diff
@@ -1,11 +1,254 @@
...
+            break;
+        default:
+            break;
+    }
+
+
+    if (!(var0 < 100)) {
+        script->varTable[0] = 1;
+    } else {
+        script->varTable[0] = 0;
```
[the remark this answers is not in the document]
Comment (Brotenko) [reply src/battle/partner/bombette_6FAD10.c:207]: I purposely don't call varTable related assignments `TRUE` or `FALSE` anymore, since there were cases before were they actually looked at the number, or suddenly a `2` appeared in another function that called it. 
tl;dr I rather have `0` and `1` instead of calling them `TRUE` and `FALSE` if I am not 100% sure.

[the remark this answers is not in the document]
Comment (nanaian) [reply src/battle/partner/bombette_6FAD10.c:207]: i mean here there are very clearly two cases
